README


This archive contains the distribution Rubric,
version 0.155:

  a notes and bookmarks manager with tagging

This software is copyright (c) 2004 by Ricardo SIGNES.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.014.



Changes
Revision history for Rubric

0.155     2014-03-20 18:48:28-04:00 America/New_York
        - improve chances of producing valid XHTML
        - allow the instance to be renamed with "basename"

0.154     2013-10-15 22:30:06 America/New_York
          update bugtracker and repo links to point at GitHub

0.153     2012-12-24 23:17:19 America/New_York
          use Email::Sender, not Email::Send (breaks smtp_server config
          for now)

0.152     2012-07-10 16:29:38 America/New_York
          add a missing prereq, stop losing errors doing session decrypt
          (thanks for the report, Jan Dubois!)

0.151     2012-02-24 10:18:29 America/New_York

          switched from YAML to YAML::XS

          now built with Dist::Zilla; some docs may be a bit weird until I go
          through them with a fine-toothed comb

0.150     2012-02-24 10:09:47 America/New_York

          TOTALLY BUSTED; DO NOT USE

0.149     2011-10-23

          more properly handle encoding for entry text/title/description --
          still a mess, but much, much more likely to just work, now


0.148     2011-08-22

          fix a bug that prevented `rubric user --new-user` from working

          include everything needed to run under PSGI -- although app.psgi
          is not currently installed anywhere

0.147     2011-02-02

          [MIGHT BREAK EVERYTHING]
          remove Config::Auto and stop using it; might break everything!

0.146     2011-01-08
          do not allow Encode to destructively process forms

0.145     2010-01-11
          do not require users to have JSON::XS

0.144     2009-03-11
          add entry title to <title> tag (bricas)
          use String::TagString

0.143     2007-07-01
          minor testing/packages updates
          add calendar view (jcap)
          add tag cloud (jcap)
          do work to make things work under fcgi (rjbs)

0.142     2006-09-05
          update install instructions

0.141     2006-09-02
          freshen Module::Install

0.140     2006-05-18
          new version numbering, new release schedule: less stable, more often
          fix a small RSS bug that bugged a large number Firefox users
          lots of forgotten changes

0.12      2005-10-26 (long overdue)
          some tiny test cleanups that might make tests... pass!
          moved repository to svn
          fixed stupid handling of repost to URIs with one-entry-per-uri config
          delete unneeded sessions at teardown (lousy?)
          work on unbreaking tags/facets
          rolled back Data::FormValidator use
          provide link on related tags to add/remove user constraint
          show related tags even when no user selected
          QUESTIONABLE DESCISION: @-prefix tags don't list in tagboxes
          list user's name for "your tags" if user ne current_user
          begin stubbing out pluggable body rendering
          begin stubbing out facets
          allow query for "/exact_tags/" to work (find untagged)
          re-allow "+" as tag separator in tagstrings

0.10      2005-06-01
          a little more and better testing
          fixed broken 'help' runmode

0.09_09   2005-05-29
          enable utf-8 tags
          fix missing-title error on blank new post form
          simplify updatedb script, eliminate vacuumdb script
          display_localtime option

0.09_08   2005-05-28
          private_tag implementation no longer completely disables public entries
          expand/collapse link now preserves page number
          help page's bookmarklet now uses encodeURIComponent, not escape
          added an input box for searching

0.09_07   2005-05-27
          private_tag implementation no longer horribly breaks entry listing
          fixes to work with DBD::SQLite 1.08, now required

0.09_06   2005-05-27
          added private_tag - SO SETTLE DOWN, SSORICHE!

0.09_05   2005-05-25
          fixed bug that snuck back into _04, reset_password was broken
          introduced Rubric::DBI::Setup for db maintenance

0.09_04   2005-05-24
          refactor get_* to return property, not WebApp
          add stubby password-reset functionality

0.09_03   2005-05-02
          fix tag-destroying bug

0.09_02   2005-04-29
          basic experimental searching
          hyphens allowed as non-first char in tag
          uri_root defaults to "" not undef

0.09_01   2005-04-12
          describe queries (badly)
          redirect to previous query after deleting a link
          validate that post input is utf-8
          add first_only query constraint
          fix RSS more: use rdf:about attr as intended!
          taglists tags are now "click to add" on post form
          added "tags for this link"
          if posting a non-unique uri, previous post defaults in (old bug?)

0.08      2005-04-07
          use normal must-login/redirect-on-login behavior for /preferences
          honor requirement for verification on unverified users
          use html_line_breaks in RSS feed for paragraphing
          POD corrections

0.07_07   2005-04-04
          added exact_tags query string
          +/- filter on taglist centers on gecko; gecko bug must die
          updated "related tags" taglist for clarity of add/remove tags
          RSS should now validate and work everywhere (I think)
          expand/collapse entry bodies on default entries templates
          the numeric tag "0" now works
          the (edit) links will now include a redirect to your previous query
          /edit or /entry links to missing entries (or non-owned for edit)
            produce better error messages
          max/default entries per page is now configurable

0.07_06   2005-04-02
          duplicate tags are now dropped; they no longer break posting
          report bad uri/tags at post-time and reproduce form
          keep long_form param on next/prev page links
          username and password for db connection can be specified in config

0.07_05   2005-03-31

Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.014.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "a notes and bookmarks manager with tagging",
  "AUTHOR" => "Ricardo SIGNES <rjbs\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.03"
  },
  "DISTNAME" => "Rubric",
  "EXE_FILES" => [
    "bin/rubric"
  ],
  "LICENSE" => "perl",
  "NAME" => "Rubric",
  "PREREQ_PM" => {
    "App::Cmd" => 0,
    "App::Cmd::Command" => 0,
    "App::Cmd::Command::commands" => 0,
    "CGI::Application" => 3,
    "CGI::Carp" => 0,
    "CGI::Cookie" => 0,
    "Carp" => 0,
    "Class::Accessor" => 0,
    "Class::DBI" => "0.96",
    "Class::DBI::AbstractSearch" => 0,
    "Class::DBI::utf8" => 0,
    "Crypt::CBC" => 0,
    "Crypt::Rijndael" => 0,
    "DBI" => 0,
    "Date::Span" => "1.12",
    "DateTime" => 0,
    "Digest::MD5" => 0,
    "Email::Address" => 0,
    "Email::Sender::Simple" => 0,
    "Encode" => 2,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "HTML::CalendarMonth" => 0,
    "HTML::TagCloud" => 0,
    "HTML::Widget::Factory" => "0.03",
    "JSON" => 2,
    "LWP::Simple" => 0,
    "MIME::Base64" => 0,
    "Scalar::Util" => 0,
    "String::TagString" => 0,
    "String::Truncate" => 0,
    "Sub::Exporter" => 0,
    "Template" => "2.00",
    "Template::Filters" => 0,
    "Template::Plugin::Class" => "0.12",
    "Time::Piece" => 0,
    "YAML::XS" => 0,
    "base" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "DBD::SQLite" => "1.08",
    "Exporter" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Copy" => 0,
    "File::Path" => 0,
    "File::Spec::Functions" => 0,
    "HTTP::Server::Simple" => "0.08",
    "HTTP::Server::Simple::CGI" => 0,
    "List::Util" => 0,
    "Test::File::ShareDir" => 0,
    "Test::HTTP::Server::Simple" => "0.02",
    "Test::More" => "0.96",
    "Test::WWW::Mechanize" => "1.04",
    "lib" => 0,
    "version" => 0
  },
  "VERSION" => "0.155",
  "test" => {
    "TESTS" => "t/*.t t/www/*.t"
  }
);


my %FallbackPrereqs = (
  "App::Cmd" => 0,
  "App::Cmd::Command" => 0,
  "App::Cmd::Command::commands" => 0,
  "CGI::Application" => 3,
  "CGI::Carp" => 0,
  "CGI::Cookie" => 0,
  "Carp" => 0,
  "Class::Accessor" => 0,
  "Class::DBI" => "0.96",
  "Class::DBI::AbstractSearch" => 0,
  "Class::DBI::utf8" => 0,
  "Crypt::CBC" => 0,
  "Crypt::Rijndael" => 0,
  "DBD::SQLite" => "1.08",
  "DBI" => 0,
  "Date::Span" => "1.12",
  "DateTime" => 0,
  "Digest::MD5" => 0,
  "Email::Address" => 0,
  "Email::Sender::Simple" => 0,
  "Encode" => 2,
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Copy" => 0,
  "File::Path" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "File::Spec::Functions" => 0,
  "HTML::CalendarMonth" => 0,
  "HTML::TagCloud" => 0,
  "HTML::Widget::Factory" => "0.03",
  "HTTP::Server::Simple" => "0.08",
  "HTTP::Server::Simple::CGI" => 0,
  "JSON" => 2,
  "LWP::Simple" => 0,
  "List::Util" => 0,
  "MIME::Base64" => 0,
  "Scalar::Util" => 0,
  "String::TagString" => 0,
  "String::Truncate" => 0,
  "Sub::Exporter" => 0,
  "Template" => "2.00",
  "Template::Filters" => 0,
  "Template::Plugin::Class" => "0.12",
  "Test::File::ShareDir" => 0,
  "Test::HTTP::Server::Simple" => "0.02",
  "Test::More" => "0.96",
  "Test::WWW::Mechanize" => "1.04",
  "Time::Piece" => 0,
  "YAML::XS" => 0,
  "base" => 0,
  "lib" => 0,
  "parent" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


